api/wasm/cpp/proxy_wasm_intrinsics.h

```diff
@@ -988,30 +991,43 @@ struct Metric : public MetricBase {
   template <typename... Fields> void record(uint64_t value, Fields... tags);
   template <typename... Fields> uint64_t get(Fields... tags);
   template <typename... Fields> uint32_t resolve(Fields... tags);
+  template <typename... Fields> Metric partiallyResolve(Fields... tags);
 };
 
-inline uint32_t MetricBase::resolveWithFields(const std::vector<const std::string>& fields) {
-  if (fields.size() != tags.size()) {
-    throw ProxyException("metric fields.size() != tags.size()");
-  }
-  size_t s = 0;
-  for (auto& t : tags) {
-    s += t.name.size() + 1; // 1 more for "."
+inline std::string MetricBase::prefixWithFields(const std::vector<const std::string>& fields) {
+  size_t s = prefix.size();
+  for (int i = 0; i < fields.size(); i++) {
+    s += tags[i].name.size() + 1; // 1 more for "."
   }
   for (auto& f : fields) {
     s += f.size() + 1; // 1 more for "."
   }
-  s += name.size() + 2; // "." and "\0";
+  s += 1; // "\0";
```

Comment:
Why do we need `\0` in `std::string`?

Reply:
In C++ 11, the space for the zero is implicit. In C++ pre-11 it is up to the implementation. 

Since we are post c++11 I can remove this.

Done.